### PR TITLE
TST/CLN: Parameterize test in sparse/test_indexing

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

### CODE REVIEW

**Comment by gfyoung** [comment]
I expect to have Azure fail until #23182 is merged.

File: pandas/tests/sparse/test_indexing.py [hunk 1]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+        arr = pd.SparseArray([1, 2, 3], kind=kind)
+        indexer = pd.SparseArray([True, False, True],
+                                 fill_value=fill,
+                                 dtype=bool)
+
+        tm.assert_sp_array_equal(pd.SparseArray([1, 3], kind=kind),
+                                 arr[indexer],)
+
+        s = pd.SparseSeries(arr, index=['a', 'b', 'c'],
+                            dtype=np.float64)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]: I think you can fit this on a single line now that it's unindented

File: pandas/tests/sparse/test_indexing.py [hunk 2]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+    @pytest.mark.parametrize("fill", [True, False, np.nan])
+    def tests_indexing_with_sparse(self, kind, fill):
+        # see gh-13985
+        arr = pd.SparseArray([1, 2, 3], kind=kind)
+        indexer = pd.SparseArray([True, False, True],
+                                 fill_value=fill,
+                                 dtype=bool)
+
+        tm.assert_sp_array_equal(pd.SparseArray([1, 3], kind=kind),
+                                 arr[indexer],)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]: Can you use a result/expected setup here?  Might make it a little more readable.
```python
result = ...
expected = ...
tm.assert_sp_array_equal(result, expected)
```

File: pandas/tests/sparse/test_indexing.py [hunk 3]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+
+        s = pd.SparseSeries(arr, index=['a', 'b', 'c'],
+                            dtype=np.float64)
+
+        exp = pd.SparseSeries(
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]: exp --> expected

```suggestion
        expected = pd.SparseSeries(
```

File: pandas/tests/sparse/test_indexing.py [hunk 4]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+            [1, 3], index=['a', 'c'],
+            dtype=SparseDtype(np.float64, s.fill_value),
+            kind=kind
+        )
+        tm.assert_sp_series_equal(s[indexer], exp)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]:
```suggestion
        tm.assert_sp_series_equal(s[indexer], expected)
```

File: pandas/tests/sparse/test_indexing.py [hunk 5]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+        tm.assert_sp_series_equal(s.loc[indexer], exp)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]:
```suggestion
        tm.assert_sp_series_equal(s.loc[indexer], expected)
```

File: pandas/tests/sparse/test_indexing.py [hunk 6]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+        tm.assert_sp_series_equal(s.iloc[indexer], exp)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]:
```suggestion
        tm.assert_sp_series_equal(s.iloc[indexer], expected)
```

File: pandas/tests/sparse/test_indexing.py [hunk 7]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+
+        indexer = pd.SparseSeries(indexer, index=['a', 'b', 'c'])
+        tm.assert_sp_series_equal(s[indexer], exp)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]:
```suggestion
        tm.assert_sp_series_equal(s[indexer], expected)
```

File: pandas/tests/sparse/test_indexing.py [hunk 8]
```diff
@@ -424,38 +424,38 @@ def test_reindex_nearest(self):
...
+        tm.assert_sp_series_equal(s.loc[indexer], exp)
```
Comment (jschendel) [inline pandas/tests/sparse/test_indexing.py]:
```suggestion
        tm.assert_sp_series_equal(s.loc[indexer], expected)
```

**Review by jschendel — COMMENTED** [review]
LGTM in general.  A few small comments (also want to try out this new "suggestion" feature).

**Comment by jschendel** [comment]
> My only gripe with the feature is that it is going to fire new commits every time.

Yeah, and doesn't look like there's a way to do multi-line edits.  Probably good if there's a single one line change that needs to be made, but otherwise might not be worthwhile atm.

>Each commit (at least for this repository) is going to trigger CI IIUC.

Is CI smart enough to cancel the job if a newer commit has been made?  I've seen it do so in some similar scenarios before, but not sure what the exact logic around that is.

**Review by jschendel — APPROVED** [review]

**Comment by jreback** [comment]
CI do cancel older builds if newer ones appear
though only if they didn’t start

**Comment by gfyoung** [comment]
> CI do cancel older builds if newer ones appear
though only if they didn’t start

Ideally, though Circle was having issues with that...

**Comment by gfyoung** [comment]
@jreback : Addressed all comments, and all is green again!  PTAL.

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**